api/public/public_api/endpoints/device_activate.py

```diff
@@ -32,64 +32,98 @@
         else:
-            response = '', HTTPStatus.NOT_FOUND
+            response = "", HTTPStatus.NOT_FOUND
+
         return response
 
-    def _validate_request(self):
+    def _validate_request(self) -> dict:
+        """Validate the contents of the API request against the data model."""
         activation_request = ActivationRequest(self.request.json)
         activation_request.validate()
 
-        return activation_request
+        return activation_request.to_native()
 
     def _get_pairing_session(self):
         """Get the pairing session from the cache.
 
         The request must have same state value as that stored in the
         pairing session.
         """
-        pairing_session_token = self.request.json['token']
+        pairing_session_token = self.request.json["token"]
         pairing_session_key = DEVICE_PAIRING_TOKEN_KEY.format(
             pairing_token=pairing_session_token
         )
         pairing_session = self.cache.get(pairing_session_key)
         if pairing_session:
             pairing_session = json.loads(pairing_session)
-            if self.request.json['state'] == pairing_session['state']:
+            if self.request.json["state"] == pairing_session["state"]:
                 self.cache.delete(pairing_session_key)
-                return pairing_session
 
-    def _activate(self, device_id: str, activation_request: ActivationRequest):
-        """Updates the core version, platform and enclosure_version columns"""
+        return pairing_session
+
+    def _activate(self, device_id: str, activation_request: dict):
+        """Updates the core version, platform and enclosure_version columns
+
+        :param device_id: internal identifier of the device
+        :param activation_request: validated request data
+        """
         updates = dict(
-            platform=str(activation_request.platform),
-            enclosure_version=str(activation_request.enclosureVersion),
-            core_version=str(activation_request.coreVersion)
+            platform=str(activation_request["platform"]),
+            enclosure_version=str(activation_request["enclosure_version"]),
+            core_version=str(activation_request["core_version"]),
         )
-        DeviceRepository(self.db).update_device_from_core(device_id, updates)
+        self.device_repository.update_device_from_core(device_id, updates)
+
+        if activation_request["pantacor_device_id"] is not None:
+            self._add_pantacor_config(
+                device_id, activation_request["pantacor_device_id"]
+            )
+
+    def _add_pantacor_config(self, device_id: str, pantacor_device_id: str):
+        """The software updates are managed by Pantacor, get their ID and add to DB
+
+        :param device_id: internal identifier of the device
+        :param pantacor_device_id: identifier of a device on the Pantacor system
+        """
+        pantacor_config = get_pantacor_device(pantacor_device_id)
```

Comment:
Do we need to catch the potential `PantacorError` in the event that the `pantacor_device_id` is not found?

Reply:
Good idea.  if the device id is passed from the device, that would imply it should be on their system

Reply:
Just looked at the code.  If you look at selene.api.pantacor the PantacorError is raised there.

Reply:
It's raised, but do we handle that scenario?
Agree that it _should_ be on their system, but unlike the previous API call approach, we're now relying on the device and Pantacor API to be in sync.
Eg what happens if I modify the device_id on my device? (I have no idea why someone would do this)